src/main/java/org/jfree/chart/plot/XYPlot.java

```diff
@@ -3986,9 +3986,8 @@ public void handleClick(int x, int y, PlotRenderingInfo info) {
      *
      * @return A list of datasets.
      */
-    private List<XYDataset<S>> getDatasetsMappedToDomainAxis(Integer axisIndex) {
+    private List<XYDataset<S>> getDatasetsMappedToDomainAxis(Integer axisIndex, List<XYDataset<S>> result) {
```

Comment:
Not sure about efficiency, but IMHO, this makes the contract of these methods worse. Previously these were pure functions without side effects, but now there are side effects introduced. I would be pretty careful here if any performance improvement warrants changing the contract of the methods like this.

Reply:
Got it!

 If such changes cause some concerns, I will undo these changes in the next commit.